### DISCUSSION TITLE
Docker install

### DISCUSSION THREAD

**Discussion by DPBiff** [TracksApp/tracks#3027] [Q&A] [answered]
I have tried to install Tracks several ways. I've managed to install it on my own linux box exactly one time.  Ruby is very difficult.  I have it running in a Turnkey appliance and it's great, but... I want to be able to access it from the cloud & can't becuse .. ISP.  So I have a VPS i want to install it on.  I'm following the instructions and can't get past the first step. This is what I get:
=> ERROR [27/28] COPY .git /app/.git                                                                                             0.0s
------
 > [27/28] COPY .git /app/.git:
------
Dockerfile:39
--------------------
  37 |     COPY db /app/db/
  38 |
  39 | >>> COPY .git /app/.git
  40 |
  41 |     RUN RAILS_ENV=production bundle exec rake assets:precompile
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref 75e4bac4-034c-4eb8-b7f2-22a96b33957d::810xvjt2hbr92ueqapobps4f4: "/.git": not found

VPS is runing Alma Linux 8, Docker is installed correctly... ?? Help?

**Comment by ZeiP** [comment, marked as answer]
There is indeed a bug in the Dockerfile, it expects to be wrunning in a Git repository and not a version archive. The easiest way to fix this is to remove the `COPY .git /app/.git` line from Dockerfile. Another option is to use a Git clone instead of a version archive by cloning the Git repository first with `git clone https://github.com/TracksApp/tracks.git`. I opened an issue #3028 for fixing this properly.

**Reply by DPBiff** [reply]
One step forward, 2 steps back:
First option - cloning Git repository: same error
2nd option -  
=> ERROR [ 6/27] RUN gem install bundler                                159.2s
------                                                                          
 > [ 6/27] RUN gem install bundler:                                             
91.25 ERROR:  Could not find a valid gem 'bundler' (>= 0), here is why:         
91.25           Unable to download data from https://rubygems.org/ - Socket::ResolutionError: Failed to open TCP connection to rubygems.org:443 (getaddrinfo: Temporary failure in name resolution) (https://rubygems.org/specs.4.8.gz)
------
Dockerfile:11
--------------------
   9 |     
  10 |     COPY Gemfile* /app/
  11 | >>> RUN gem install bundler
  12 |     RUN bundle install --jobs 4
  13 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c gem install bundler" did not complete successfully: exit code: 2

**Reply by ZeiP** [reply]
You seem to be having some problems with your Docker network setup, Bundler is not able to access the repository. Unfortunately I don't have any ideas why that is, but it seems like that is not an issue related to Tracks but more widely to your Docker setup. 

Which instructions are you using, BTW? Building the image yourself isn't necessary unless you make changes to the code / configuration files (and often not even then). Just noticed that the current installation instructions don't cover this; basically you can skip the [build step](https://github.com/TracksApp/tracks/blob/master/doc/installation.md) and use the Docker hub image instead by replacing the Docker commands with
`docker run --link mariadb:db --rm -t tracksapp/tracks bin/rake db:reset`
and
`docker run -d -p 3000:3000 --name tracks --link mariadb:db -t tracksapp/tracks`

**Reply by DPBiff** [reply]
First line gives me:
docker run --link mariadb:db --rm -t tracksapp/tracks bin/rake db:reset
docker: Error response from daemon: could not get container for mariadb: No such container: mariadb.
See 'docker run --help'.

**Reply by ZeiP** [reply]
Did you follow the installation instructions I linked?

**Reply by DPBiff** [reply]
Yes I'm following them.  I'm trying 2 different approaches: Trying to get docker to work on my VSP & doing a manual install on a new VM

**Reply by ZeiP** [reply]
If you have followed the instructions, there should be a postgres or mariadb container present, which doesn't seem to be the case. You can check that via docker ps.

**Reply by DPBiff** [reply]
Ok here's the result of 'docker ps'
──╼ \>docker ps
CONTAINER ID   IMAGE              COMMAND                  CREATED        STATUS        PORTS      NAMES
b8d8ed808a60   tracksapp/tracks   "/app/docker-entrypo…"   40 hours ago   Up 40 hours   3000/tcp   romantic_williamson

**Reply by ZeiP** [reply]
Yes, so you don't have any database containers running. Check the installation instructions to verify you've actually launched them properly. Self-running Tracks does require knowledge of Docker etc. and I don't have the resources to give give individual Docker support.

**Reply by DPBiff** [reply]
Well I ditched docker as it doesn't seem to be any less difficult than a manual installation.  After going through the posted installation instructions, correcting the errors, and figuring out how to install things that don't work, I have Tracks running on a VM (Alma Linux) however, the first instance of it returns an 500 error.  I'm hoping you can take a peek at the lot (attached) and tell me how to proceed.
[production.log](https://github.com/user-attachments/files/16050126/production.log)
